odpf/shield/v1beta1/shield.proto

```diff
@@ -644,6 +656,24 @@ message CreateUserResponse {
   User user = 1;
 }
 
+message MetadataKeyRequestBody {
+  string key = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
```

Comment:
How about key only having Upper Caps, like enum?

Reply:
This regex shall allow all uppercase keys. 
`ex: STOP, START etc would be valid`

I think it's the desired behavior(?)

Reply:
It will also allow - `Edit`, `slack-Id`

Reply:
@krtkvrm Can you please help in listing down the formats/examples of keys that should be accepted and not accepted?

Reply:
Should we just let me plain string? Any specific need to put validation on this apart from string?

Reply:
No, just regex that would be acceptable for key
Like `STOP_FIREHOSE` or `Stop-Firehose` or `StopFirehose`.. etc. I would prefer 1st one, open to discussions

Reply:
I wanted to keep a check on the use of some symbols that generally are not there in keys. Examples of such keys are` @type, stop(), 1==1`. By allowing pain text, users can use anything for the key. 

Shall we still allow plain text @ravisuhag ?

Reply:
@krtkvrm Yes, shall we allow all of these, including all small caps and snake case?  Ex `manager, manager_email`

Reply:
+1. I think we can just skip symbols and allow all other characters.